src/OrchardCore/OrchardCore.Configuration.KeyVault/Extensions/AzureKeyVaultWebHostBuilderExtension.cs

```diff
@@ -16,16 +17,36 @@ public static class AzureKeyVaultWebHostBuilderExtension
         /// <returns>The web host builder.</returns>
         public static IHostBuilder AddOrchardCoreAzureKeyVault(this IHostBuilder builder)
         {
-            if (builder == null) throw new ArgumentNullException(nameof(builder));
+            if (builder == null) 
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            builder.AddOrchardCoreAzureKeyVault(new DefaultAzureCredential(includeInteractiveCredentials: true));
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Adds Azure Key Vault as a Configuration Source.
+        /// </summary>
+        /// <param name="builder">The web host builder to configure.</param>
+        /// <param name="tokenCredential">The token credential to use for authentication.</param>
+        /// <returns>The web host builder.</returns>
+        public static IHostBuilder AddOrchardCoreAzureKeyVault(this IHostBuilder builder, TokenCredential tokenCredential)
```

Comment:
@deanmarcussen Cool but i assume you need to use the `tokenCredential` below line 63

Reply:
Yes I do!

And I will mark it not ready, as I need to check it provides what was not working tonight